### PR TITLE
CBRN hose support

### PR DESCRIPTION
**Pull request by Kexanone**
**When merged this pull request will:**
- connect CBRN hoses for exported SQF, vanilla arsenal and deserialized units.
- not fix it for ACE arsenal. This should be done on ACE's side.

### CODE REVIEW

File: addons/common/functions/fnc_openArsenal.sqf [hunk 1]
```diff
@@ -23,5 +23,18 @@ if (GVAR(preferredArsenal) == 1 && {isClass (configFile >> "CfgPatches" >> "ace_
 
     [_unit, _unit, true] call ace_arsenal_fnc_openBox;
 } else {
+    GVAR(arsenalCenter) = _unit;
+
+    // Connect CBRN hoses when arsenal is closed
+    if (isNil QGVAR(arsenalCbrnEH)) then {
+        GVAR(arsenalCbrnEH) = [
+            missionnamespace,
```
Comment (CreepPork) [inline addons/common/functions/fnc_openArsenal.sqf:31]: missionNamespace

File: addons/common/functions/fnc_exportMissionSQF.sqf [hunk 2]
```diff
@@ -114,6 +114,7 @@ private _fnc_processInventory = {
 
     if (_object isKindOf "CAManBase") then {
         _outputObjects pushBack ["%1 setUnitLoadout %2;", _varName, getUnitLoadout _object];
+        _outputObjects pushBack ["%1 call BIN_fnc_CBRNHoseInit;", _varName];
```
Comment (neilzar) [inline addons/common/functions/fnc_exportMissionSQF.sqf:117]: The incorrect facemask is returned by `getUnitLoadout`, leaving the unit without a respirator when the SQF is executed but the hose still exists.

Comment (neilzar) [reply addons/common/functions/fnc_exportMissionSQF.sqf:117]: Maybe a frame delay will fix this issue?

Comment (neilzar) [reply addons/common/functions/fnc_exportMissionSQF.sqf:117]: Checking again the mask appears for one frame and is then removed for some reason.

Comment (oOKexOo) [reply addons/common/functions/fnc_exportMissionSQF.sqf:117]: Yes, `BIN_fnc_CBRNHoseInit` and `setUnitLoadout` needs to be executed in the next frame. Dunno how I missed it.

Comment (oOKexOo) [reply addons/common/functions/fnc_exportMissionSQF.sqf]: Should be resolved now

**Review by neilzar — APPROVED** [review]

File: addons/common/functions/fnc_exportMissionSQF.sqf [hunk 3]
```diff
@@ -110,11 +110,17 @@ private _fnc_processGroup = {
...
-
     if (_object isKindOf "CAManBase") then {
-        _outputObjects pushBack ["%1 setUnitLoadout %2;", _varName, getUnitLoadout _object];
+        _outputObjects pushBack "[{";
+        _outputObjects pushBack "    params [""_unit""];";
+        if !(_object call FUNC(hasDefaultInventory)) then {
+            _outputObjects pushBack ["    _unit setUnitLoadout %1;", getUnitLoadout _object];
+        };
+        _outputObjects pushBack "    _unit call BIN_fnc_CBRNHoseInit;";
+        _outputObjects pushBack ["}, [%1]] call CBA_fnc_execNextFrame;", _varName];
```
Comment (mharis001) [inline addons/common/functions/fnc_exportMissionSQF.sqf:120]: I think we should decide whether we want export mission SQF to require any mods (CBA in this case) to work. If so, we could then change portions of the code to just call ZEN function as well.

Comment (Kexanone) [reply addons/common/functions/fnc_exportMissionSQF.sqf:120]: I forgot about that. I can easily swap it with `BIS_fnc_addStackedEventHandler`.

Comment (CreepPork) [inline addons/common/functions/fnc_exportMissionSQF.sqf:120]: What's with the spaces?

Comment (Kexanone) [reply addons/common/functions/fnc_exportMissionSQF.sqf:120]: Proper indentation?

**Pull request merged**